tests/integration-tests/tests/update/test_update.py

```diff
@@ -529,3 +529,159 @@ def _test_max_vcpus(region, stack_name, vcpus):
+
+    # Update cluster with new configuration and change the compute instance type to t2.xlarge
+    new_compute_instance_type = "t2.xlarge"
+    _update_config_compute_instance_type(
+        init_config, pcluster_config_reader, test_datadir, cluster, new_compute_instance_type, init_config_name
+    )
+
+    _check_cfn_scheduler_slots(cluster, scheduler, region, new_compute_instance_type)
+
+
+def _check_extra_json_update_compute(
+    pcluster_config_reader, clusters_factory, init_config_name, test_datadir, scheduler, region, cfn_scheduler_slots
+):
+    init_config_file = pcluster_config_reader(config_file=init_config_name)
+    cluster = clusters_factory(init_config_file)
+
+    init_config = configparser.ConfigParser()
+    init_config.read(init_config_file)
+
+    # Update cluster with new configuration and change the compute instance type to c5.9xlarge
+    new_compute_instance_type = "c5.9xlarge"
+    _update_config_compute_instance_type(
+        init_config, pcluster_config_reader, test_datadir, cluster, new_compute_instance_type, init_config_name
+    )
+
+    _check_cfn_scheduler_slots(
+        cluster, scheduler, region, new_compute_instance_type, cfn_scheduler_slots=cfn_scheduler_slots
+    )
+
+
+def _update_config_compute_instance_type(
+    init_config, pcluster_config_reader, test_datadir, cluster, new_compute_instance_type, file_name
+):
+    init_config.set("cluster default", "compute_instance_type", new_compute_instance_type)
+    updated_config_file = pcluster_config_reader(config_file=_save_config_file(init_config, test_datadir, file_name))
+    cluster.config_file = str(updated_config_file)
+    cluster.update()
+
+
+def _save_config_file(config, test_datadir, file_name):
+    try:
+        with open(str(test_datadir / file_name), "w") as configfile:
+            config.write(configfile)
+        return file_name
+    except IOError as err:
+        logging.info("I/O error: {0}".format(err))
+
+
+def _check_cfn_scheduler_slots(
+    cluster, scheduler, region, new_compute_instance_type, disable_hyperthreading=False, cfn_scheduler_slots=None
+):
+    # Command executors
+    command_executor = RemoteCommandExecutor(cluster)
+    scheduler_commands = get_scheduler_commands(scheduler, command_executor)
+
+    # Get new slots_per_instance
+    slots_per_instance = fetch_instance_slots(region, new_compute_instance_type)
+    if disable_hyperthreading or cfn_scheduler_slots == "cores":
+        slots_per_instance = slots_per_instance // 2
+    elif cfn_scheduler_slots and cfn_scheduler_slots.isdigit():
+        slots_per_instance = int(cfn_scheduler_slots)
+
+    # submit a job to check cfn_scheduler_slot has been updated
+    result = scheduler_commands.submit_command("sleep 1", nodes=1, slots=slots_per_instance)
+    job_id = scheduler_commands.assert_job_submitted(result.stdout)
+    scheduler_commands.wait_job_completed(job_id)
+    assert_that(scheduler_commands.compute_nodes_count()).is_equal_to(1)
```

Comment:
can we also assert the actual number of slots seen by the scheduler?

Reply:
what do you mean actual number of slots seen by the scheduler

Reply:
no I mean running the qhost command and check how many slots are configured in the scheduler for the specific node

Reply:
Done